### DISCUSSION TITLE
Serveur Auth not running

### DISCUSSION THREAD

**Discussion by hyvan29444** [gravitational/teleport#36043] [Q&A] [answered]
hello everyone, hello @webvictim , please i need some help.

I'm trying to set up the HA for the authentication service, but it's not working.
I have this ERROR when i'm trying to see my auth server:  **Cannot connect to the auth server. Is the auth server running on "0.0.0.0:3025"?**
I use postgre for the storage type and if I disable this configuration line, the authentication service works again. 
here's my configuration file (please note that the connection to the database is not secure, I'm still testing)
Thanks for your help

> teleport:
>   nodename: auth
>   advertise_ip: *******
>   log:
>     output: stderr
>     severity: DEBUG
>   data_dir: /var/lib/teleport
>   storage:
>     type: postgresql
>     conn_string: postgresql://user:pass@bdd IP/teleport_backend?sslmode=disable&pool_max_conns=20
>     change_feed_conn_string: postgresql://user:Pass@bdd IP/teleport_backend?sslmode=disable
>     audit_events_uri:
>       - postgresql://user:pass@bdd IP/teleport_audit?sslmode=disable#disable_cleanup=false&retention_period=2160h
> 
> auth_service:
>   enabled: "yes"
>   keep_alive_interval: 1m
>   keep_alive_count_max: 3
>   listen_addr: 0.0.0.0:3025
>   public_addr: auth_IP:3025
>   proxy_listener_mode: multiplex
>   authentication:
>     second_factor: off
>   cluster_name: ha-proxy-teleport
>   
> ssh_service:
>   enabled: no
> 
> proxy_service:
>   enabled: no
>

**Comment by hyvan29444** [comment]
I have the same problem when i'm trying with **etcd** as type of storage.

Teleport version: v14.2.3

**Comment by webvictim** [comment]
This would suggest that there's an error caused by that PostgreSQL config which causes Teleport to fail to start. What's in your Teleport logs at that time? `sudo journalctl -u teleport --no-pager`

**Comment by hyvan29444** [comment]
Hi, thanks for your reply. 
This is the output of the command

 sudo journalctl -u teleport --no-pager --since "2024-01-11 12:20"
janv. 11 12:20:00 nte-auth-1 teleport[6168]: 2024-01-11T12:20:00+01:00 INFO [PGBK]      Setting up change feed. slot_name:teleport_cb35acc297274599b0fa78f147cdf565 pgbk/background.go:187
janv. 11 12:20:00 nte-auth-1 teleport[6168]: 2024-01-11T12:20:00+01:00 INFO [PGBK]      Change feed started. slot_name:teleport_cb35acc297274599b0fa78f147cdf565 pgbk/background.go:201
janv. 11 12:20:00 nte-auth-1 teleport[6168]: 2024-01-11T12:20:00+01:00 INFO [PGEVENTS]  Setting up events backend. pgevents/pgevents.go:196
janv. 11 12:20:02 nte-auth-1 teleport[6168]: 2024-01-11T12:20:02+01:00 INFO [PGEVENTS]  Started events backend. pgevents/pgevents.go:222
janv. 11 12:20:02 nte-auth-1 teleport[6168]: ERROR REPORT:
janv. 11 12:20:02 nte-auth-1 teleport[6168]: Original Error: *trace.BadParameterError please specify audit_sessions_uri when using external audit backends
janv. 11 12:20:02 nte-auth-1 teleport[6168]: Stack Trace:
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         github.com/gravitational/teleport/lib/service/service.go:1624 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthExternalAuditLog
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         github.com/gravitational/teleport/lib/service/service.go:1696 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthService
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         github.com/gravitational/teleport/lib/service/service.go:1095 github.com/gravitational/teleport/lib/service.NewTeleport
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         github.com/gravitational/teleport/lib/service/service.go:679 github.com/gravitational/teleport/lib/service.newTeleportProcess
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         github.com/gravitational/teleport/lib/service/service.go:689 github.com/gravitational/teleport/lib/service.Run
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         github.com/gravitational/teleport/tool/teleport/common/teleport.go:609 github.com/gravitational/teleport/tool/teleport/common.OnStart
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         github.com/gravitational/teleport/tool/teleport/common/teleport.go:527 github.com/gravitational/teleport/tool/teleport/common.Run
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         github.com/gravitational/teleport/tool/teleport/main.go:31 main.main
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         runtime/proc.go:267 runtime.main
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         runtime/asm_amd64.s:1650 runtime.goexit
janv. 11 12:20:02 nte-auth-1 teleport[6168]: User Message: initialization failed
janv. 11 12:20:02 nte-auth-1 teleport[6168]:         please specify audit_sessions_uri when using external audit backends
janv. 11 12:20:02 nte-auth-1 systemd[1]: teleport.service: Main process exited, code=exited, status=1/FAILURE
janv. 11 12:20:02 nte-auth-1 systemd[1]: teleport.service: Failed with result 'exit-code'.
janv. 11 12:20:02 nte-auth-1 systemd[1]: teleport.service: Consumed 4.407s CPU time.
janv. 11 12:20:02 nte-auth-1 systemd[1]: teleport.service: Scheduled restart job, restart counter is at 17.
janv. 11 12:20:02 nte-auth-1 systemd[1]: Stopped Teleport Service.
janv. 11 12:20:02 nte-auth-1 systemd[1]: teleport.service: Consumed 4.407s CPU time.
janv. 11 12:20:02 nte-auth-1 systemd[1]: Started Teleport Service.
janv. 11 12:20:03 nte-auth-1 teleport[6176]: 2024-01-11T12:20:03+01:00 INFO             Starting Teleport v14.2.3 with a config file located at "/etc/teleport.yaml" common/teleport.go:607
janv. 11 12:20:03 nte-auth-1 teleport[6176]: 2024-01-11T12:20:03+01:00 DEBU [SQLITE]    Connected to: file:/var/lib/teleport/proc/sqlite.db?_busy_timeout=10000&_sync=FULL&_txlock=immediate, poll stream period: 1s lite/lite.go:254
janv. 11 12:20:03 nte-auth-1 teleport[6176]: 2024-01-11T12:20:03+01:00 DEBU [SQLITE]    journal_mode=delete, synchronous=2, busy_timeout=10000 lite/lite.go:305
janv. 11 12:20:04 nte-auth-1 teleport[6176]: 2024-01-11T12:20:04+01:00 DEBU [PROC:1]    Using postgresql backend. pid:6176.1 service/service.go:5429
janv. 11 12:20:04 nte-auth-1 teleport[6176]: 2024-01-11T12:20:04+01:00 INFO [PGBK]      Setting up backend. pgbk/pgbk.go:181
janv. 11 12:20:05 nte-auth-1 teleport[6176]: 2024-01-11T12:20:05+01:00 DEBU [BUFFER]    Add Watcher(name=external_audit_storage, prefixes=/external_audit_storage/cluster, capacity=1024, size=0). backend/buffer.go:262
janv. 11 12:20:05 nte-auth-1 teleport[6176]: 2024-01-11T12:20:05+01:00 INFO [PGBK]      Starting change feed stream. pgbk/background.go:99
janv. 11 12:20:06 nte-auth-1 teleport[6176]: 2024-01-11T12:20:06+01:00 INFO [PGBK]      Setting up change feed. slot_name:teleport_a49ddcb6c7b14a67b4e0f4196e97cf2d pgbk/background.go:187
janv. 11 12:20:06 nte-auth-1 teleport[6176]: 2024-01-11T12:20:06+01:00 INFO [PGBK]      Change feed started. slot_name:teleport_a49ddcb6c7b14a67b4e0f4196e97cf2d pgbk/background.go:201
janv. 11 12:20:06 nte-auth-1 teleport[6176]: 2024-01-11T12:20:06+01:00 INFO [PGEVENTS]  Setting up events backend. pgevents/pgevents.go:196
janv. 11 12:20:09 nte-auth-1 teleport[6176]: 2024-01-11T12:20:09+01:00 INFO [PGEVENTS]  Started events backend. pgevents/pgevents.go:222
janv. 11 12:20:09 nte-auth-1 teleport[6176]: ERROR REPORT:
janv. 11 12:20:09 nte-auth-1 teleport[6176]: Original Error: *trace.BadParameterError please specify audit_sessions_uri when using external audit backends
janv. 11 12:20:09 nte-auth-1 teleport[6176]: Stack Trace:
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         github.com/gravitational/teleport/lib/service/service.go:1624 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthExternalAuditLog
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         github.com/gravitational/teleport/lib/service/service.go:1696 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthService
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         github.com/gravitational/teleport/lib/service/service.go:1095 github.com/gravitational/teleport/lib/service.NewTeleport
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         github.com/gravitational/teleport/lib/service/service.go:679 github.com/gravitational/teleport/lib/service.newTeleportProcess
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         github.com/gravitational/teleport/lib/service/service.go:689 github.com/gravitational/teleport/lib/service.Run
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         github.com/gravitational/teleport/tool/teleport/common/teleport.go:609 github.com/gravitational/teleport/tool/teleport/common.OnStart
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         github.com/gravitational/teleport/tool/teleport/common/teleport.go:527 github.com/gravitational/teleport/tool/teleport/common.Run
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         github.com/gravitational/teleport/tool/teleport/main.go:31 main.main
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         runtime/proc.go:267 runtime.main
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         runtime/asm_amd64.s:1650 runtime.goexit
janv. 11 12:20:09 nte-auth-1 teleport[6176]: User Message: initialization failed
janv. 11 12:20:09 nte-auth-1 teleport[6176]:         please specify audit_sessions_uri when using external audit backends
janv. 11 12:20:09 nte-auth-1 systemd[1]: teleport.service: Main process exited, code=exited, status=1/FAILURE
janv. 11 12:20:09 nte-auth-1 systemd[1]: teleport.service: Failed with result 'exit-code'.
janv. 11 12:20:09 nte-auth-1 systemd[1]: teleport.service: Consumed 5.822s CPU time.
janv. 11 12:20:09 nte-auth-1 systemd[1]: teleport.service: Scheduled restart job, restart counter is at 18.
janv. 11 12:20:09 nte-auth-1 systemd[1]: Stopped Teleport Service.
janv. 11 12:20:09 nte-auth-1 systemd[1]: teleport.service: Consumed 5.822s CPU time.
janv. 11 12:20:09 nte-auth-1 systemd[1]: Started Teleport Service.

**Comment by hyvan29444** [comment]
##Database table##
psql (14.10 (Ubuntu 14.10-0ubuntu0.22.04.1), server 16.1 (Debian 16.1-1.pgdg120+1))
WARNING: psql major version 14, server major version 16.
         Some psql features might not work.
Type "help" for help.

teleport_backend=# \dt
              List of relations
 Schema |      Name       | Type  |  Owner
--------+-----------------+-------+----------
 public | backend_version | table | postgres
 public | events          | table | postgres
 public | kv              | table | postgres
 public | meta            | table | postgres
(4 rows)

**Reply by hyvan29444** [reply]
● teleport.service - Teleport Service
     Loaded: loaded (/lib/systemd/system/teleport.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2024-01-11 12:20:39 CET; 2s ago
   Main PID: 6224 (teleport)
      Tasks: 6 (limit: 1054)
     Memory: 31.4M
        CPU: 2.457s
     CGroup: /system.slice/teleport.service
             └─6224 /usr/local/bin/teleport start --config /etc/teleport.yaml --pid-file=/run/teleport.pid

janv. 11 12:20:39 nte-auth-1 systemd[1]: Started Teleport Service.
janv. 11 12:20:39 nte-auth-1 teleport[6224]: 2024-01-11T12:20:39+01:00 INFO             Starting Teleport v14.2.3 with a config file located at "/etc/teleport.yaml" c>
janv. 11 12:20:39 nte-auth-1 teleport[6224]: 2024-01-11T12:20:39+01:00 DEBU [SQLITE]    Connected to: file:/var/lib/teleport/proc/sqlite.db?_busy_timeout=10000&_sync=>
janv. 11 12:20:39 nte-auth-1 teleport[6224]: 2024-01-11T12:20:39+01:00 DEBU [SQLITE]    journal_mode=delete, synchronous=2, busy_timeout=10000 lite/lite.go:305
janv. 11 12:20:40 nte-auth-1 teleport[6224]: 2024-01-11T12:20:40+01:00 DEBU [PROC:1]    Using postgresql backend. pid:6224.1 service/service.go:5429
janv. 11 12:20:40 nte-auth-1 teleport[6224]: 2024-01-11T12:20:40+01:00 INFO [PGBK]      Setting up backend. pgbk/pgbk.go:181
janv. 11 12:20:41 nte-auth-1 teleport[6224]: 2024-01-11T12:20:41+01:00 DEBU [BUFFER]    Add Watcher(name=external_audit_storage, prefixes=/external_audit_storage/clus>
janv. 11 12:20:41 nte-auth-1 teleport[6224]: 2024-01-11T12:20:41+01:00 INFO [PGBK]      Starting change feed stream. pgbk/background.go:99
janv. 11 12:20:42 nte-auth-1 teleport[6224]: 2024-01-11T12:20:42+01:00 INFO [PGBK]      Setting up change feed. slot_name:teleport_181d73bac5eb4eb0a3e72da3b7fd3306 pg>
lines 1-19/19 (END)


root@nte-auth-1:/home/superuser# tctl auth ls
2024-01-11T12:21:03+01:00 DEBU [SQLITE]    Connected to: file:/var/lib/teleport/proc/sqlite.db?_busy_timeout=10000&_sync=FULL&_txlock=immediate, poll stream period: 1s lite/lite.go:254
2024-01-11T12:21:03+01:00 DEBU [SQLITE]    journal_mode=delete, synchronous=2, busy_timeout=10000 lite/lite.go:305
2024-01-11T12:21:04+01:00 DEBU             Connecting to: [{0.0.0.0:3025 tcp }]. authclient/authclient.go:63
ERROR: Cannot connect to the auth server. Is the auth server running on "0.0.0.0:3025"?

**Reply by webvictim** [reply, marked as answer]
The error does give good context here - `audit_sessions_uri` must also be set when using a Postgres-based backend. You should configure one of the supported HA session recording backends to make sure that session recordings are available to all auth servers using the Postgres backend:

- AWS S3 (or S3-like): https://goteleport.com/docs/reference/backends/#s3
- Google Cloud Storage: https://goteleport.com/docs/reference/backends/#gcs
- Azure Blob Storage: https://goteleport.com/docs/reference/backends/#azure-blob-storage

If you really want to use local storage (for testing) you can set this in the config and restart:

```yaml
auth_service:
  storage:
    audit_sessions_uri: "file:///var/lib/teleport/log/records"
```

Bear in mind that this should not be used for production setups or for HA, as it will result in errors.

I've raised an issue here to document this better: https://github.com/gravitational/teleport/issues/36619

**Comment by hyvan29444** [comment]
Hi 
Thank you for your support, it works very well. And yes in prod a cloud service will be used instead of local storage
